peerloop/core/utils/validation.py

```diff
@@ -0,0 +1,12 @@
+from email_validator import EmailNotValidError, validate_email
+
+from peerloop.core.exceptions.validation import InvalidEmailError
+
+
+def validate_email_format(email: str) -> str:
+    try:
+        validation = validate_email(email)
+        valid_email = validation.email
+    except EmailNotValidError:
+        raise InvalidEmailError(f"Invalid Email Error: {email} is not a valid email address.")
+    return valid_email
```

Comment:
Wouldn't it be more appropriate that this returns a boolean(success/fail) instead of the email?
1. `validate_email_format`, as its same suggests, is a validation function which usually returns a boolean.
2. The reason it used to return the same email was that it behaved like a "middleware" that intercepts the value, validates, and pass along when there's no problem. However, in this case, it's no longer a Pydantic decorator but a validation function.

Reply:
Maybe another function name like `is_valid_email_format` would suit better? I'm not also sure :(